aspnetcore/fundamentals/configuration/index.md

```diff
@@ -59,6 +59,8 @@ var builder = WebApplication.CreateBuilder(args);
 1. [appsettings.json](#appsettingsjson) using the [JSON configuration provider](#jcp).
 1. A fallback to the host configuration described in the [next section](#host).
 
+Note: `WebApplication.CreateBuilder(args)` should only be called once.
```

Comment:
It's perfectly fine to call `WebApplication.CreateBuilder(args)` multiple times if you're restarting the server, or using Kestrel on multiple ports to host multiple applications.

I also took a look at https://techcommunity.microsoft.com/blog/iis-support-blog/asp-net-core-503-server-has-been-shutdown/3830338, and really it should be okay to do the same for IIS as long as you don't start the app created by the builder, but we have a finalizer that's doing things it shouldn't.

I'll submit a PR to the aspnetcore repo to fix that and post a link to it here. It probably won't be backported though since it's not a regression and it shouldn't be necessary `WebApplication.CreateBuilder(args)` multiple times in an app hosted on IIS.

Reply:
```suggestion
Note: `WebApplication.CreateBuilder(args)` should only be called once in apps relying on IIS in-process hosting.
```